evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -139,28 +134,20 @@ def transform(self, X, y=None):
             ww.DataTable: Transformed data, where each categorical feature has been encoded into numerical columns using one-hot encoding.
         """
         X_ww = infer_feature_types(X)
-        X_copy = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
-        X_copy = self._handle_parameter_handle_missing(X_copy)
-
-        X_t = pd.DataFrame()
-        # Add the non-categorical columns, untouched
-        for col in X_copy.columns:
-            if col not in self.features_to_encode:
-                X_t = pd.concat([X_t, X_copy[col]], axis=1)
-        # The call to pd.concat above changes the type of the index so we will manually keep it the same.
-        if not X_t.empty:
-            X_t.index = X_copy.index
+        X_copy = self._handle_parameter_handle_missing(X_ww)
+
+        X_ww = X_ww.ww.drop(self.features_to_encode)
 
         # Call sklearn's transform on the categorical columns
         if len(self.features_to_encode) > 0:
             X_cat = pd.DataFrame(self._encoder.transform(X_copy[self.features_to_encode]).toarray(), index=X_copy.index)
             X_cat.columns = self._get_feature_names()
-            X_t = pd.concat([X_t, X_cat], axis=1)
-            X_t = X_t.drop(columns=self._features_to_drop)
-            self._feature_names = X_t.columns
-        booleans = {c: "Boolean" for c in self.get_feature_names()}
-        X_ww = _retain_custom_types_and_initalize_woodwork(X_ww, X_t)
-        X_ww = X_ww.set_types(booleans)
+            X_cat = X_cat.drop(columns=self._features_to_drop)
```

Comment:
No need to call `retain_custom_types` here because our implementation preserves the schema!